compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala

```diff
@@ -458,7 +458,10 @@ abstract class PrepJSInterop extends plugins.PluginComponent
 
       val isJSNative = !sym.hasAnnotation(ScalaJSDefinedAnnotation)
 
-      if (isJSNative && !isJSAnonFun &&
+      if (sym.isPackageObjectClass) {
+        reporter.warning(implDef.pos,
+            "package objects inheriting from js.Any are deprecated")
```

Comment:
- Consider adding: "Use a normal object instead."
- Add a full stop at the and
- Consider starting with upper case (our error messages seem to do so, I don't know about scalac)